### PR TITLE
optionally supports remote embeddings

### PR DESCRIPTION
**Pull request by thoraxe**
## Type of change

- [ ] Refactor
- [X] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Previously, text embedding was done locally. This adds optional support for a remote embedding server (HuggingFace TEI) via ENV.

## Related Tickets & Documents
N/A

## Checklist before requesting a review

- [X] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
Local testing.

### CODE REVIEW

**Comment by bparees** [comment]
lgtm (i have no merge perms)

**Comment by vishnuchalla** [comment]
Any thoughts around having a `constants.py` file to have all hard coded constants at one place so that they can be used across the repo?

File: tools/indexer.py [hunk 1]
```diff
@@ -11,7 +11,15 @@
 llama_index.set_global_handler("simple")
 
 # Select Model
-service_context = get_watsonx_context(model="ibm/granite-13b-chat-v1")
+## check if we are using remote embeddings via env
+tei_embedding_url = os.getenv("TEI_SERVER_URL", None)
+
+if tei_embedding_url != None:
+    service_context = get_watsonx_context(model="ibm/granite-13b-chat-v1",
```
Comment (vishnuchalla) [inline tools/indexer.py:18]: Do we want to parameterize the other arguments in the function too? just a nit maybe we can do it in our subsequent PRs.

**Comment by thoraxe** [comment]
I have no objection to a constants file. I can try to work that up.

File: tools/indexer.py (hunk 1 above)
Comment (thoraxe) [reply tools/indexer.py:18]: If you're using the TEI server, technically it ignores the `tei_embedding_model` completely. The `get_watsonx_context` function already has parameterized stuff, so I'm not sure what you're asking.

**Pull request merged**